walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -238,6 +248,99 @@
     }
 
     /**
+     * Gets all the applicant's interested jobs given the params. If there's an error in getting a particular job
+     * from the jobId, then that specific job will not be returned.
+     *
+     * @param applicantId The applicant's userId.
+     * @param pageSize The the number of jobs to be shown on the page.
+     * @param pageIndex The page number on which we are at.
+     * @return Future of the JobPage object.
+     * @throws IllegalArgumentException If the applicantId doesn't have a corresponding document.
+     */
+    public Future<JobPage> fetchInterestedJobPage(String applicantId, int pageSize, int pageIndex) throws IOException, IllegalArgumentException {
+        CollectionReference applicantAccountsCollection = FireStoreUtils.getFireStore().collection(APPLICANT_ACCOUNTS_COLLECTION);
+
+        DocumentReference docRef = applicantAccountsCollection.document(applicantId);
+
+        return ApiFutures.transform(
+            docRef.get(),
+            documentSnapshot -> {
+                if (!documentSnapshot.exists()) {
+                    throw new IllegalArgumentException("Invalid applicantId");
+                }
+
+                List<String> interestedList = (List<String>) documentSnapshot.get(INTERESTED_JOBS_FIELD);
+                // TODO(issue/34): adjust the interestedList the be looked at based on pagination
+
+                ImmutableList.Builder<Job> jobListBuilder = ImmutableList.builder();
+                try {
+                    int counter = 0;
+                    // Handles the jobIds 10 at a time.
```

Comment:
Because the 10 is due to a restriction on firestore IN query i would actually suggest you move that logic of splitting your list inside the JobsDb manager instead. so you pass the whole list to it and the db manager will break it up in batches of 10 and return the combined list.

Also replace the 10 (magic number) by a constant so you don't need to explain what it is,m something like FIRESTORE_IN_QUERY_MAX_ARGS